sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,577 @@
+                an Integer, the actual physical port connected to a Y cable
+
+        Returns:
+            an Integer, 1 if TOR A is actively linked and routing(TOR 1).
+                      , 2 if TOR B is actively linked and routing(TOR 2).
+                      , 0 if nothing linked and actively routing
+                      , -1 if checking which side linked for routing API fails.
+        """
+
+        raise NotImplementedError
+
+    def check_if_link_is_active_for_NIC(self, port):
+        """
+        This API specifically checks if NIC side of the Y cable's link is active
+
+        Args:
+            port:
+                 an Integer, the actual physical port connected to a Y cable
+
+        Returns:
+            a boolean, true if the link is active
+                     , false if the link is not active
+        """
+
+        raise NotImplementedError
+
+    def check_if_link_is_active_for_torA(self, port):
+        """
+        This API specifically checks if tor A side of the Y cable's link is active
+
+        Args:
+            port:
+                 an Integer, the actual physical port connected to a Y cable
+
+        Returns:
+            a boolean, true if the link is active
+                     , false if the link is not active
+        """
+
+        raise NotImplementedError
+
+    def check_if_link_is_active_for_torB(self, port):
+        """
+        This API specifically checks if tor B side of the Y cable's link is active
+
+        Args:
+            port:
+                 an Integer, the actual physical port connected to a Y cable
+
+        Returns:
+            a boolean, true if the link is active
+                     , false if the link is not active
+        """
+
+        raise NotImplementedError
+
+    def get_eye_info(self, port):
+        """
+        This API specifically returns the EYE height value for a specfic port.
+        The target could be local side, TOR1, TOR2, NIC etc.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+            target:
+                 an Integer, the target on which to enable the PRBS
+                             EYE_PRBS_TARGET_LOCAL -> local side,
+                             EYE_PRBS_TARGET_TOR1 -> TOR 1
+                             EYE_PRBS_TARGET_TOR2 -> TOR 2
+                             EYE_PRBS_TARGET_NIC -> NIC
+        Returns:
+            a list, with EYE values of lane 0 lane 1 lane 2 lane 3 with corresponding index
+        """
+
+        raise NotImplementedError
+
+    def get_ber_info(self, port):
+        """
+        This API specifically returns the BER (Bit error rate) value for a specfic port.
+        The target could be local side, TOR1, TOR2, NIC etc.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+            target:
+                 an Integer, the target on which to enable the PRBS
```

Comment:
Maybe rather than explicitly stating "An Integer", we simply state "One of the following predefined constants:". This would also allow us to change the underlying type transparently if desired.

Reply:
fixed